clients/src/main/java/org/apache/kafka/clients/NodeApiVersions.java

```
@@ -173,23 +187,68 @@ private String apiVersionToText(ApiVersion apiVersion) {
         }
 
         if (apiKey != null) {
-            Short usableVersion = usableVersions.get(apiKey);
-            if (usableVersion == NODE_TOO_OLD)
+            UsableVersion usableVersion = usableVersions.get(apiKey);
+            if (usableVersion.isTooOld())
                 bld.append(" [unusable: node too old]");
-            else if (usableVersion == NODE_TOO_NEW)
+            else if (usableVersion.isTooNew())
                 bld.append(" [unusable: node too new]");
             else
-                bld.append(" [usable: ").append(usableVersion).append("]");
+                bld.append(" [usable: ").append(usableVersion.value).append("]");
         }
         return bld.toString();
     }
 
     public ApiVersion apiVersion(ApiKeys apiKey) {
-        for (ApiVersion nodeApiVersion : nodeApiVersions) {
-            if (nodeApiVersion.apiKey == apiKey.id) {
-                return nodeApiVersion;
+        UsableVersion usableVersion = usableVersions.get(apiKey);
+        if (usableVersion == null)
+            throw new NoSuchElementException();
```

Can we mention the requested `apiKey` in the exception? Also, it would be good to document that the method throws an exception instead of just returning `null` if no `apiVersion` can be found.